plugins/basic/nexus-timeline-plugin/src/main/java/org/sonatype/nexus/timeline/feeds/sources/SystemFeedSource.java

```diff
@@ -46,11 +47,54 @@ public SystemFeedSource(final FeedRecorder feedRecorder) {
 
   @Override
   public void fillInEntries(final List<FeedEvent> entries, final int from, final int count,
-                            final Map<String, String> params)
+                            final Map<String, Object> params)
   {
-    entries.addAll(getFeedRecorder()
-        .getEvents(ImmutableSet.of(FeedRecorder.FAMILY_SYSTEM, FeedRecorder.FAMILY_REPO), null, from, count,
-            filters(params)
-        ));
+    entries.addAll(
+        getFeedRecorder().getEvents(
+            ImmutableSet.of(FeedRecorder.FAMILY_SYSTEM, FeedRecorder.FAMILY_REPO),
+            null,
+            toWhere(params),
+            params,
+            from,
+            count,
+            new Function<FeedEvent, FeedEvent>()
+            {
+              @Override
+              public FeedEvent apply(final FeedEvent input) {
+                input.setTitle(title(input));
+                return input;
+              }
+            }
+        )
+    );
+  }
+
+  /**
+   * Formats entry title for event.
+   */
+  private String title(FeedEvent evt) {
+    if (FeedRecorder.SYSTEM_BOOT.equals(evt.getEventSubType())) {
+      return "Nexus " + evt.getData().get("bootAction");
+    }
+    else if (FeedRecorder.SYSTEM_CONFIG.equals(evt.getEventSubType())) {
+      return "Configuration change";
+    }
+    else if (FeedRecorder.FAMILY_REPO.equals(evt.getEventType())) {
+      if (FeedRecorder.REPO_CREATED.equals(evt.getEventSubType())) {
+        return "Repository created";
+      }
+      else if (FeedRecorder.REPO_UPDATED.equals(evt.getEventSubType())) {
+        return "Repository updated";
+      }
+      else if (FeedRecorder.REPO_DELETED.equals(evt.getEventSubType())) {
+        return "Repository deleted";
+      } else {
+        return evt.getEventType() + ":" + evt.getEventSubType();
```

Comment:
better in this case to init this as a default and just have a final return statement with it instead of duplicating the logic across cases